undercrawler/spiders/base_spider.py

```diff
@@ -362,3 +357,25 @@ def link_to_url(link):
     if link.fragment and link.fragment != '#':
         return "#".join([link.url, link.fragment])
     return link.url
+
+
+def allowed_re(url, hard_url_constraint):
```

Comment:
Yes, I think it makes sense to allow all subdomains lower then the start subdomain, I'll update the regexp.